### ISSUE TITLE
Bug: grad(sum(inv(A))) wrong result -- TriangularSolve JVP missing `triu(dA)` mask? (v0.1.9)

### ISSUE THREAD

**Issue by olli4** [ekzhang/jax-js#84]
grad(sum(inv(A))) gives the wrong result.

## JAX

```python
import jax
import jax.numpy as jnp
import jax.scipy.linalg as jax_linalg

A = jnp.array([[2, 1], [1, 3]], dtype=jnp.float32)

def f(a):
  inv_a = jnp.linalg.inv(a)
  return inv_a.sum()

jax_grad = jax.grad(f)(A)
print(jax_grad)
```

Output:

```text
[[-0.16000001 -0.08      ]
 [-0.08000001 -0.04      ]]
```

## jax-js

```TS
import { grad, numpy as np } from "@jax-js/jax";

const A = np.array([[2, 1], [1, 3]], { dtype: "float32" });

const f = (a) => {
  const inv_a = np.linalg.inv(a);
  return inv_a.sum();            
};

const dA = grad(f)(A.ref);
console.log(dA.js());
```

Actual output:

```text
[
  [ -0.5649999976158142, -0.23000001907348633 ],
  [ -0.07000000029802322, -0.14000000059604645 ]
]
```

Expected output: same as JAX

AI says that the bug is that the JVP rule for TriangularSolve neglects to mask `da` it passes to `batchMatmulT(da, x.ref)` as `unitDiagonal ? triu(da, 1) : triu(da)`. In my fork this solved the problem.

**Comment by ekzhang** [comment]
Thank you for the bug report
